scripts/ansible-todd-dev.yml

```diff
@@ -1,83 +1,135 @@
 ---
 - name: Sudo-level VM Prep
   hosts: all
-  become: yes
+  become: true
   become_user: root
   vars:
-    todd_dir: /home/vagrant/go/src/github.com/Mierdin/todd
-    gover: 1.6.2  #Change this variable to update to a new version of Go
-    gosha256: e40c36ae71756198478624ed1bb4ce17597b3c19d243f3f0899bb5740d56212a
+    docker_repo: 'deb https://apt.dockerproject.org/repo {{ ansible_distribution|lower }}-{{ ansible_distribution_release|lower }} main'
+    todd_dir: '/home/vagrant/go/src/github.com/Mierdin/todd'
+    todd_make_configureenv_check_file: '/var/log/.make_configureenv_complete'
+    gover: '1.6.2'  #Change this variable to update to a new version of Go
+    gosha256: 'e40c36ae71756198478624ed1bb4ce17597b3c19d243f3f0899bb5740d56212a'
   tasks:
 
-  - name: Set up docker repo
-    apt_repository:
-      repo='deb https://apt.dockerproject.org/repo ubuntu-trusty main'
-      state=present
-
-  - name: Install APT dependencies
-    apt: pkg={{ item }} state=latest update_cache=yes force=yes  # Currently forcing because of a certificate error for docker-engine
-    with_items:
-    - git
-    - docker-engine
-
-  - name: Add vagrant user to docker group
-    user: name=vagrant groups="docker" append=yes
-
-  - name: Ensure docker service is restarted
-    service: name=docker state=restarted
-
-  - name: Download Go tarball
-    get_url:
-      url="https://storage.googleapis.com/golang/go{{ gover }}.linux-amd64.tar.gz"
-      checksum="sha256:{{ gosha256 }}"
-      validate_certs=no
-      dest=/tmp/go{{ gover }}.linux-amd64.tar.gz
-      mode=0770
-
-  - name: Unzip Go tarball
-    unarchive:
-      src=/tmp/go{{ gover }}.linux-amd64.tar.gz
-      dest=/usr/local
-      copy=no
-
-  - name: Config Files
-    file:
-      src={{ todd_dir }}/etc
-      dest=/etc/todd
-      state=link
-
-  - name: Configure Environment
-    command: make configureenv chdir={{ todd_dir }}
-
-  - name: GOPATH Owner
-    shell: chown vagrant:vagrant $(find /home/vagrant/go | grep -v todd) warn=no
+    - name: Set up docker repo
+      apt_repository:
+        repo: '{{ docker_repo }}'
+        state: "present"
+
+    - name: Install APT dependencies
+      apt:
+        name: "{{ item }}"
+        state: "latest"
+        update_cache: yes
+        force: yes  # Currently forcing because of a certificate error for docker-engine
+      with_items:
+        - 'git'
+        - 'docker-engine'
+
+    - name: Add vagrant user to docker group
+      user:
+        name: "vagrant"
+        groups: "docker"
+        append: yes
+      register: vagrant_user_added_to_docker_group
+
+    - name: Ensure docker service is restarted
+      service:
+        name: "docker"
+        state: "restarted"
+      when: vagrant_user_added_to_docker_group.changed
+
+    - name: Download Go tarball
+      get_url:
+        url: "https://storage.googleapis.com/golang/go{{ gover }}.linux-amd64.tar.gz"
```

Comment:
For consistency, could this not also be placed in vars? My only question would be since a var is already being used to determine the version to download, I assume you'd have to declare the ```gover``` variable first, then a second variable that contains this whole URL, with the ```gover``` variable embedded. I can't say I remember ever trying this (declaring a var based on a var that was declared in the same play)